crates/hc_service_check/src/main.rs

```diff
@@ -0,0 +1,355 @@
+use clap::{Parser, Subcommand};
+use std::io::{Error, Result};
+
+#[derive(Subcommand, Debug)]
+enum Cmd {
+    /// Check the health of a bootstrap server.
+    Bootstrap {
+        /// The url of the bootstrap server to check.
+        #[arg(short, long, default_value = "https://bootstrap.holo.host")]
+        url: String,
+    },
+
+    /// Check the health of a signal server.
+    Signal {
+        /// The url of the signal server to check.
+        #[arg(short, long, default_value = "wss://signal.holo.host")]
+        url: String,
+    },
+
+    /// Check the health of a turn server.
+    Turn {
+        /// The host of the turn server to check.
+        #[arg(short = 'e', long, default_value = "turn.holo.host")]
+        host: String,
+
+        /// The port of the turn server to check.
+        #[arg(short, long, default_value_t = 443)]
+        port: u16,
+
+        /// The username of the turn server to check.
+        #[arg(short, long, default_value = "hc")]
+        user: String,
+
+        /// The credential of the turn server to check.
+        #[arg(short, long, default_value = "h20240315")]
+        cred: String,
+
+        /// The transport of the turn server to check.
+        #[arg(short, long, default_value = "udp")]
+        transport: String,
+    },
+}
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[command(subcommand)]
+    cmd: Cmd,
+}
+
+#[tokio::main(flavor = "multi_thread")]
+async fn main() {
+    let Args { cmd } = Args::parse();
+
+    match match cmd {
+        Cmd::Bootstrap { url } => bootstrap(url).await,
+        Cmd::Signal { url } => signal(url).await,
+        Cmd::Turn {
+            host,
+            port,
+            user,
+            cred,
+            transport,
+        } => turn(host, port, user, cred, transport).await,
+    } {
+        Ok(()) => println!("done."),
+        Err(err) => eprintln!("{err:?}"),
+    }
+}
+
+async fn bootstrap(url: String) -> Result<()> {
+    println!("boostrap check of {url}");
+    println!("checking 'now' command");
+    let now = kitsune_p2p_bootstrap_client::now(
+        Some(url2::Url2::parse(url)),
+        kitsune_p2p_bootstrap_client::BootstrapNet::Tx5,
+    )
+    .await
+    .map_err(Error::other)?;
+    println!("got 'now' result: {now}");
+    Ok(())
+}
+
+async fn signal(url: String) -> Result<()> {
+    println!("signal check of {url}");
+    let (cli, _rcv) = tx5_signal::Cli::builder()
+        .with_url(url2::Url2::parse(url).into())
+        .build()
+        .await
+        .unwrap();
+    let ice = cli.ice_servers();
+    println!("got signal connect result: {ice:#?}");
+    Ok(())
+}
+
+const ONE_KB: [u8; 1024] = [0xdb; 1024];
+const MSG_CNT: usize = 2;
```

Comment:
i might be useful to make this a CLI argument for bandwidth testing

Reply:
done: https://github.com/holochain/holochain/pull/3496/commits/fc1a54cf2b0255785957eb038bcc6ed1bf6e347c